src/main/java/com/heeverse/security/JwtAuthenticationFilter.java

```diff
@@ -0,0 +1,38 @@
+package com.heeverse.security;
+
+import com.heeverse.security.JwtTokenProvider;
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.util.ObjectUtils;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+/**
+ * @author jeongheekim
+ * @date 2023/08/02
+ */
+public class JwtAuthenticationFilter extends OncePerRequestFilter {
+
+    public static final String HEADER_NAME = "Authorization";
+    private final JwtTokenProvider jwtTokenProvider;
+
+    public JwtAuthenticationFilter(JwtTokenProvider jwtTokenProvider) {
+        this.jwtTokenProvider = jwtTokenProvider;
+    }
+
+    @Override
+    protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response,
+        FilterChain filterChain) throws ServletException, IOException {
+
+        String jwtToken = jwtTokenProvider.parsing(request.getHeader(HEADER_NAME));
+        if (!ObjectUtils.isEmpty(jwtToken) && jwtTokenProvider.validateToken(jwtToken)) {
```

Comment:
`!ObjectUtils.isEmpty(jwtToken) && jwtTokenProvider.validateToken(jwtToken)`과 같이 조건이 중첩된 경우, 어떤것을 검증하고자 하는지 이름을 붙여주는것이 좋습니다.

변수로 변경하셔도 좋고, 메서드로 변경하셔도 좋습니다. 코드를 읽을 때 이 코드의 의도를 명확하게 파악할 수 있어야 하기 때문에, 중첩된 조건문이나 복잡한 조건의 경우에는 의도를 담은 이름을 지어주시는건 어떠신가요?

```suggestion
        if (isValidJsonToken(jwtToken)) {
```

Reply:
중첩된 조건문에 대해 변수로 빼서 어떠한 조건을 검사하지 말 수 있도록 수정하겠습니다.
또한 isEmpty메소드를 validateToken 메서드로 빼서 조건문을 간소화해 볼 수도 있을 거 같습니다.